Create integrating-with-laravel.md

richeklein opened the pull request:
Instructions for using Shoelace components in a Laravel 8.x application.

richeklein committed:
Create integrating-with-laravel.md

Instructions for using Shoelace components in a Laravel 8.x application.

claviska commented on docs/tutorials/integrating-with-laravel.md:
```diff
@@ -0,0 +1,85 @@
+### Configure Laravel Mix
+[Laravel Mix](https://laravel.com/docs/8.x/mix) is a wrapper around Webpack that simplifies configuration. Mix is used by default for compiling front-end assets in Laravel.
+
+Modify `webpack.mix.js` to add Shoelace's assets to Webpack's build process
```
We probably want to make sure `<sl-icon>` works using the default icon library (components that use icons internally use the `system` library, which doesn't require assets). You can test this by adding `<sl-icon name="house"></sl-icon>`. It probably won't render anything, so we'll need to add a call to `setBasePath` as described here:

https://shoelace.style/getting-started/installation?id=setting-the-base-path

richeklein replied:
Sure. The house icon displayed without explicitly setting the basepath. I was only able to get the icons working again using `setBasePath("/")` -- does that sound right?

claviska replied:
Looking at line 46, it should probably be set to `/assets`. The base path basically needs to point to the same folder the assets get copied to on the server (but translated to URL).

richeklein replied:
I initially tried `setBasePath("/assets");`, but that doesn't resolve properly: 
`/assets/assets/icons/house.svg:1 Failed to load resource: the server responded with a status of 404 (Not Found)`

Maybe it's the way I configured the copy in `webpack.mix.js`: 
`.copy("node_modules/@shoelace-style/shoelace/dist/assets", "public/assets")`
The entire set of Shoelace icons ends up in `/public/assets/icons`.

claviska commented on docs/tutorials/integrating-with-laravel.md:
````diff
@@ -0,0 +1,85 @@
+Import each Shoelace component you plan to use in `/resources/js/boostrap.js`:
+
+```js
+import { SlAlert, SlAnimation, SlButton, ... } from '@shoelace-style/shoelace'
````
Components are registered via side effects and I'm pretty sure webpack will load the entire library even if you only import a single component this way (i.e. using the bare module specifier). I'd love to be wrong about this but, last I checked, the behavior was how I described it here.

It's unfortunate, but we may need to [cherry pick using the full path](https://shoelace.style/getting-started/installation?id=cherry-picking) for each component to prevent a bloated webpack bundle.

richeklein replied:
Looks like Webpack still behaves the same way. Using the full path for each component cut the bundle size of the `vendor.js` in half -- at least in my example app.

claviska replied:
It's probably worth updating the imports then — and maybe calling out that a copy/paste link for components is available at the bottom of each docs page.

richeklein replied:
Done. Let me know if you see the updates.

claviska reviewed (changes requested):
This is great! I left a couple comments on things to verify before this goes live. Did you want to add a link to the sidebar under "Tutorials" as well?

richeklein committed:
Update integrating-with-laravel.md

Added a section for setting the base path and switched to using the full import path for each component. Also included a full `bootstrap.js` example.

claviska reviewed (approved):
Thank you!
